From 53eae29e8900af2df96236669dc8487dde0afb4b Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Tue, 19 Feb 2019 20:15:41 +0900
Subject: [PATCH 1/3] controller: remove python2 support

This update is to remove python2 supportfor the reasons.

* SPP CLI support both of python2 and 3, but spp-ctl does not support
  python2.

* Include all of required packages in `requirements.txt` for python3.

* Python2 will not be maintained after 2020.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Signed-off-by: Itsuro Oda <oda@valinux.co.jp>
---
 src/controller/commands/pri.py     | 2 --
 src/controller/plugins/hello.py    | 2 +-
 src/controller/shell.py            | 2 --
 src/controller/shell_lib/common.py | 2 +-
 src/controller/spp.py              | 4 +---
 src/controller/spp_common.py       | 2 +-
 src/controller/spp_ctl_client.py   | 2 +-
 src/controller/websocket/spp_ws.py | 2 +-
 src/spp.py                         | 3 +--
 9 files changed, 7 insertions(+), 14 deletions(-)

diff --git a/src/controller/commands/pri.py b/src/controller/commands/pri.py
index 2e3c208..7043c2f 100644
--- a/src/controller/commands/pri.py
+++ b/src/controller/commands/pri.py
@@ -1,8 +1,6 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2018 Nippon Telegraph and Telephone Corporation
 
-from __future__ import absolute_import
-
 from .. import spp_common
 from ..shell_lib import common
 from ..spp_common import logger
diff --git a/src/controller/plugins/hello.py b/src/controller/plugins/hello.py
index 905f476..f0189f5 100644
--- a/src/controller/plugins/hello.py
+++ b/src/controller/plugins/hello.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2017-2018 Nippon Telegraph and Telephone Corporation
 
diff --git a/src/controller/shell.py b/src/controller/shell.py
index f95dbfa..21543d1 100644
--- a/src/controller/shell.py
+++ b/src/controller/shell.py
@@ -1,8 +1,6 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2015-2019 Intel Corporation
 
-from __future__ import absolute_import
-
 import cmd
 from .commands import bye
 from .commands import pri
diff --git a/src/controller/shell_lib/common.py b/src/controller/shell_lib/common.py
index 447584b..4941bba 100644
--- a/src/controller/shell_lib/common.py
+++ b/src/controller/shell_lib/common.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2017-2018 Nippon Telegraph and Telephone Corporation
 
diff --git a/src/controller/spp.py b/src/controller/spp.py
index a74209b..00ffd6f 100644
--- a/src/controller/spp.py
+++ b/src/controller/spp.py
@@ -1,9 +1,7 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2015-2016 Intel Corporation
 
-from __future__ import absolute_import
-
 import argparse
 import re
 from .shell import Shell
diff --git a/src/controller/spp_common.py b/src/controller/spp_common.py
index 09a6b5a..6bf8b1f 100644
--- a/src/controller/spp_common.py
+++ b/src/controller/spp_common.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2015-2016 Intel Corporation
 
diff --git a/src/controller/spp_ctl_client.py b/src/controller/spp_ctl_client.py
index 50cbfae..3cce628 100644
--- a/src/controller/spp_ctl_client.py
+++ b/src/controller/spp_ctl_client.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2018 Nippon Telegraph and Telephone Corporation
 
diff --git a/src/controller/websocket/spp_ws.py b/src/controller/websocket/spp_ws.py
index 848945a..de5feeb 100755
--- a/src/controller/websocket/spp_ws.py
+++ b/src/controller/websocket/spp_ws.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2017-2018 Nippon Telegraph and Telephone Corporation
 
diff --git a/src/spp.py b/src/spp.py
index 62606e7..498fd2e 100755
--- a/src/spp.py
+++ b/src/spp.py
@@ -1,8 +1,7 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2015-2016 Intel Corporation
 
-from __future__ import absolute_import
 from controller import spp
 import sys
 
-- 
2.17.1

